WcaOnRails/app/views/competitions/_competition_form.html.erb

```diff
@@ -252,8 +252,8 @@
           function UpdateDues(){
 
             var obj = $("#competition-form").serializeJSON();
-            var compLimitEnabled = obj['competition[competitor_limit_enabled]']
-            var compLimit = compLimitEnabled === "true" ? obj['competition[competitor_limit]'] : null
+            var compLimitEnabled = ("true" === obj['competition[competitor_limit_enabled]']) // Coercing this variable to be a boolean, as one expects.
```

Comment:
Comment seems a bit unnecessary. Also, why is this specific change needed?

Reply:
In general I'm a huge fan of over-commenting, especially on a shared codebase. I can edit it to make it more clear if needed - but essentially `obj['competition[competitor_limit_enabled]']` returns a string value of `"true"`, instead of a bool `true` as one would intuitively expect from a variable named compLimitEnabled.

The issue is if someone writes `if (compLimitEnabled) {...`, it will always return true. Thus, I want to force compLimitEnabled to be a boolean when it is assigned, so that it behaves as expected throughout the code if anyone makes changes to this file later on.